### DISCUSSION TITLE
Dependency overriding not working

### DISCUSSION THREAD

**Discussion by nksabertooth** [fastapi/fastapi#10044] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
#### Trying to override get_connection_with_db method.

class PostgresUtility():
   def get_connection_with_db(self,auto_commit,db_name: str):
        connection = psycopg2.connect(host=***, port=**, user=***, password=****, dbname=****)

..................................................................
### code for override

def override_get_db(self,auto_commit,db_name: str):
    try:
        db = TestingSessionLocal()
        yield db
    finally:
        db.close()


app.dependency_overrides[PostgresUtility.get_connection_with_db] = override_get_db

client = TestClient(app)
```

### Description

I'm trying to run some tests with pytest against the API but my dependency override for database connection is not working. It still tries to run the function as if there was no override. 

### Operating System

Windows

### FastAPI Version

0.65.2

### Pydantic Version

1.9.1

### Python Version

3.10.4

**Comment by jgould22** [comment, marked as answer]
Are you using pytest?

Personally I have had better luck using pytests monkeypatch fixture for things like this. 

Additionally, 0.65.2 is a pretty old version of fastapi, the first thing I would always do is update it to the latest version as there has been many changes between that and 100.1

**Reply by nksabertooth** [reply]
Yes, I am using pytest. Can you please share a sample code for overriding the db connection using monkeypatch

**Reply by nksabertooth** [reply]
@jgould22 thanks monkeypatch worked!

**Reply by jgould22** [reply]
Sorry I was unable to reply quicker, glad you solved your problem!

**Reply by nksabertooth** [reply]
@jgould22  MonkePatch is not working for dependency in a router endpoint.
example: 
import someutility

some_utility_object = someutility()
@router.get("/col_list")
def get_col_list(request: Request, response: Response, Authorization: str = Header(None),
                         value: dict = Depends(some_utility_object.some_validation_fun)):

Any pointers on how to override  some_validation_fun?

**Reply by nksabertooth** [reply]
Finally, fastapi dependency override functionality worked here!
